### ISSUE TITLE
Error when trying to open font: "openstep_plist.parser.ParseError: Unexpected EOF while parsing plist"

### ISSUE THREAD

**Issue by arrowtype** [googlefonts/glyphsLib#795]
I have the following script:

```py
from glyphsLib import GSFont

print(GSFont)

glyphs_file = "/Users/stephennixon/type-repos/shantell-sans/sources/shantellsans-wght_ital_IRGL.glyphs"

font =  GSFont(glyphs_file)

print(font)
```

And it outputs the following:

```console
<class 'glyphsLib.classes.GSFont'>
Traceback (most recent call last):
  File "/Users/stephennixon/type-repos/shantell-sans/notes/2022_07_15-glyphsLib-userData-tests/glyphs-userdata.py", line 8, in <module>
    font =  GSFont(glyphs_file)
  File "/Users/stephennixon/type-repos/shantell-sans/venv/lib/python3.10/site-packages/glyphsLib/classes.py", line 4310, in __init__
    p.parse_into_object(self, openstep_plist.load(fp, use_numbers=True))
  File "src/openstep_plist/parser.pyx", line 561, in openstep_plist.parser.load
  File "src/openstep_plist/parser.pyx", line 544, in openstep_plist.parser.loads
  File "src/openstep_plist/parser.pyx", line 505, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 417, in openstep_plist.parser.parse_plist_dict
  File "src/openstep_plist/parser.pyx", line 397, in openstep_plist.parser.parse_plist_dict_content
  File "src/openstep_plist/parser.pyx", line 507, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 367, in openstep_plist.parser.parse_plist_array
  File "src/openstep_plist/parser.pyx", line 505, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 417, in openstep_plist.parser.parse_plist_dict
  File "src/openstep_plist/parser.pyx", line 397, in openstep_plist.parser.parse_plist_dict_content
  File "src/openstep_plist/parser.pyx", line 507, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 367, in openstep_plist.parser.parse_plist_array
  File "src/openstep_plist/parser.pyx", line 505, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 417, in openstep_plist.parser.parse_plist_dict
  File "src/openstep_plist/parser.pyx", line 397, in openstep_plist.parser.parse_plist_dict_content
  File "src/openstep_plist/parser.pyx", line 507, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 367, in openstep_plist.parser.parse_plist_array
  File "src/openstep_plist/parser.pyx", line 505, in openstep_plist.parser.parse_plist_object
  File "src/openstep_plist/parser.pyx", line 417, in openstep_plist.parser.parse_plist_dict
  File "src/openstep_plist/parser.pyx", line 397, in openstep_plist.parser.parse_plist_dict_content
  File "src/openstep_plist/parser.pyx", line 501, in openstep_plist.parser.parse_plist_object
openstep_plist.parser.ParseError: Unexpected EOF while parsing plist
```

It acts as expected if I instead point to a different Glyphs file, simply outputting:

```console
<class 'glyphsLib.classes.GSFont'>
<GSFont "Familyname">
```

I have updated to glyphsLib `6.0.6` and verified this from the same script, in the same venv:

```py
import glyphsLib
print(glyphsLib.__version__)
```

It seems things are getting hung up on the following step, but I’m not quite sure how to clarify what exactly the parser is tripping up on.

https://github.com/googlefonts/glyphsLib/blob/97642ea5e755e92567a6a3c7470c6306f8a4df3d/Lib/glyphsLib/classes.py#L4318

I assume there may be an issue in my Glyphs source. How might I find those issues?

If it’s relavant: I converted to a Glyphs source starting from UFOs, and GlyphsApp itself doesn’t report any errors. So, there also seems to be the chance that it isn’t an error in my Glyphs file. 

Any insights are greatly appreciated!

**Comment by simoncozens** [comment]
There's some binary data in there, which the openstep-plist parser doesn't handle.

**Comment by anthrotype** [comment]
openstep-plist library can parse plist binary data but it has to be wrapped with `<` and `>` and only contain hexadecimal digits, that's who openstep plist format works AFAIK, unless Glyphs.app deviates from this?

**Comment by anthrotype** [comment]
see https://github.com/fonttools/openstep-plist/blob/12b5f55d8634326669d23d3e01dbdd2fd8285b59/tests/test_parser.py#L232-L243

**Comment by arrowtype** [comment]
Thanks so much for helping me interpret that error!

One guess is that it might be image data... I’ll see if I can find and remove any images that might be lurking in the file. 

Aside from images, is there other binary data which might be found in a Glyphs file?

**Comment by anthrotype** [comment]
can you upload (or share privately) the source file? I'd like to see how the offending line looks like

**Comment by arrowtype** [comment]
Ugh, well ... I don't know what I changed, but I can no longer reproduce this issue. I can't find anything that has changed via my project Git history. I don't think I've changed anything meaningful in my system or venv's Python configuration. I did have to restart my computer at some point, and maaybe that somehow helped? I'm slightly relieved but mostly confused. Closing this issue, and will reopen if I end up reproducing this again.
